ts/features/bonus/siciliaVola/screens/voucherList/VoucherListScreen.tsx

```diff
@@ -0,0 +1,204 @@
+    <ActivityIndicator
+      color={"black"}
+      accessible={false}
+      importantForAccessibility={"no-hide-descendants"}
+      accessibilityElementsHidden={true}
+      testID={"activityIndicator"}
+    />
+  </>
+);
+
+const EmptyVoucherList = () => {
+  const dispatch = useIODispatch();
+
+  return (
+    <>
+      <InfoScreenComponent
+        image={<EmptyListImage width={104} height={104} />}
+        title={I18n.t("bonus.sv.voucherList.emptyList.title")}
+        body={I18n.t("bonus.sv.voucherList.emptyList.subtitle")}
+      />
+      <FooterWithButtons
+        type={"SingleButton"}
+        leftButton={confirmButtonProps(
+          () => dispatch(svGenerateVoucherStart()),
+          I18n.t("bonus.sv.voucherList.emptyList.cta")
+        )}
+      />
+    </>
+  );
+};
+
+const VoucherListScreen = (props: Props): React.ReactElement => {
+  const { showAnimatedModal, hideModal } = useContext(LightModalContext);
+
+  const vouchers = toArray(props.indexedVouchers);
+
+  useEffect(() => {
+    props.requestVoucherState();
+    props.resetFilter();
+  }, []);
+
+  useEffect(() => {
+    if (isUndefined(props.requiredDataLoaded)) {
+      props.requestVoucherPage(props.filters);
+    }
+  }, [props.filters, props.requiredDataLoaded]);
+
+  const openFiltersModal = () =>
+    showAnimatedModal(
+      <SvVoucherListFilters onClose={hideModal} onConfirm={hideModal} />,
+      BottomTopAnimation
+    );
+
+  return (
+    <BaseScreenComponent
+      goBack={true}
+      contextualHelp={emptyContextualHelp}
+      headerTitle={I18n.t("bonus.sv.headerTitle")}
+      isSearchAvailable={{ enabled: true, onSearchTap: openFiltersModal }}
+    >
+      <SafeAreaView style={IOStyles.flex} testID={"VoucherListScreen"}>
+        {!isReady(props.possibleVoucherState) ? (
+          <LoadingErrorComponent
+            isLoading={isLoading(props.possibleVoucherState)}
+            loadingCaption={I18n.t("bonus.sv.voucherList.loading")}
+            onRetry={props.requestVoucherState}
+          />
+        ) : (
+          <>
+            <H1 style={IOStyles.horizontalContentPadding}>
+              {I18n.t("bonus.sv.voucherList.title")}
+            </H1>
+            <View spacer />
+
+            {isReady(props.requiredDataLoaded) && vouchers.length === 0 ? (
+              <EmptyVoucherList />
+            ) : (
+              <FlatList
+                style={[IOStyles.horizontalContentPadding]}
+                data={vouchers}
+                ListFooterComponent={
+                  isLoading(props.requiredDataLoaded) && <FooterLoading />
+                }
+                keyExtractor={v => v.idVoucher.toString()}
+                ItemSeparatorComponent={() => (
+                  <ItemSeparatorComponent noPadded={true} />
+                )}
+                onEndReached={() => {
+                  if (props.uiParameters.nextPage !== undefined) {
+                    props.requestVoucherPage(props.filters);
+                  }
```

Comment:
If the call to `/beneficiario/ricercaVoucher` fails during the scroll of the list the user doesn't receive any feedback. Maybe it might be better to show him an error toast?